text/0000-iri-v181-transaction-and-bundle/0000-iri-v181-transaction-and-bundle.md

```diff
@@ -0,0 +1,980 @@
+| `timestamp`                       | the time when the transaction was issued                | 27              | `add_message` or `add_input` or `add_output`  |
+| `current_index`                   | the index of the transaction in its bundle              | 27              | `finalize`                                    |
+| `last_index`                      | the index of the last transaction in the bundle         | 27              | `finalize`                                    |
+| `bundle`                          | the hash of the bundle essence                          | 243             | `finalize`                                    |
+| `trunk`                           | the hash of the first transaction referenced/approved   | 243             | `calculate_proof_of_work`                     |
+| `branch`                          | the hash of the second transaction referenced/approved  | 243             | `calculate_proof_of_work`                     |
+| `tag`                             | arbitrary user-defined value                            | 81              | `add_message` or `add_input` or `add_output`  |
+| `attachment_timestamp`            | the timestamp for when Proof-of-Work is completed       | 27              | `calculate_proof_of_work`                     |
+| `attachment_timestamp_lowerbound` | *not specified*                                         | 27              | `calculate_proof_of_work`                     |
+| `attachment_timestamp_upperbound` | *not specified*                                         | 27              | `calculate_proof_of_work`                     |
+| `nonce`                           | the Proof-of-Work nonce of the transaction              | 81              | `calculate_proof_of_work`                     |
+
+Each transaction is uniquely identified by its *transaction hash*, which is calculated based on all fields of the
+transaction. Note that the transaction hash is not part of the transaction.
+
+Because modifying the fields of a transaction would invalidate its hash, `Transaction` is immutable after creation.
+
+### `Transaction` struct
+
+A transaction is represented by the `Transaction` struct and follows the structure presented in the table above:
+
+```rust
+pub struct Transaction {
+    signature_or_message_fragment: SignatureOrMessageFragment,
+    address: Address,
+    value: Value,
+    obsolete_tag: Tag,
+    timestamp: Timestamp,
+    current_index: Index,
+    last_index: Index,
+    bundle: BundleHash,
+    trunk: TransactionHash,
+    branch: TransactionHash,
+    tag: Tag,
+    attachment_timestamp: Timestamp,
+    attachment_timestamp_lower_bound: Timestamp,
+    attachment_timestamp_upper_bound: Timestamp,
+    nonce: Nonce
+}
+```
+
+The `Transaction` type only contains two constructor methods getter methods for read-only borrows of its fields.
```

Comment:
- two constructor functions and some getter methods for read-only borrows of its fields

Reply:
Fixed and removed one getter method. The implementation phase should define more constructors.